triage/experiments/validate.py

```diff
@@ -0,0 +1,221 @@
+                    raise ValueError(
+                        'choice query does not run. \nchoice query: "{}"\nFull error: {}'
+                        .format(categorical['choice_query'], e)
+                    )
+
+    def _validate_from_obj(self, from_obj):
+        conn = self.db_engine.connect()
+        logging.info('Validating from_obj')
+        try:
+            conn.execute('explain select * from {}'.format(from_obj))
+        except Exception as e:
+            raise ValueError(
+                'from_obj query does not run. \nfrom_obj: "{}"\nFull error: {}'
+                .format(from_obj, e)
+            )
+
+    def _validate_time_intervals(self, intervals):
+        logging.info('Validating time intervals')
+        for interval in intervals:
+            if interval != 'all':
+                _ = convert_str_to_relativedelta(interval)
+
+    def _validate_groups(self, groups):
+        if not any(group == 'entity_id' for group in groups):
+            raise ValueError('entity_id needs to be present in each feature_aggregation\'s list of groups')
+
+    def _validate_aggregation(self, aggregation_config):
+        logging.info('Validating aggregation config %s', aggregation_config)
+        self._validate_keys(aggregation_config)
+        self._validate_aggregates(aggregation_config)
+        self._validate_categoricals(aggregation_config.get('categoricals', []))
+        self._validate_from_obj(aggregation_config['from_obj'])
+        self._validate_time_intervals(aggregation_config['intervals'])
+        self._validate_groups(aggregation_config['groups'])
+
+    def run(self, feature_aggregation_config):
+        """Validate a feature aggregation config applied to this object
+
+        The validations range from basic type checks, key presence checks,
+        as well as validating the sql in from objects.
+
+        Args:
+            feature_aggregation_config (list) all values, except for feature
+                date, necessary to instantiate a collate.SpacetimeAggregation
+
+        Raises: ValueError if any part of the config is found to be invalid
+        """
+        if not feature_aggregation_config:
+            raise ValueError("configuration key 'feature_aggregations' not found")
+        for aggregation in feature_aggregation_config:
+            self._validate_aggregation(aggregation)
+
+class EventsTableValidator(Validator):
+    def run(self, events_table):
+        if not events_table:
+            raise ValueError("configuration key 'events_table' not found")
+        table_should_have_data(events_table, self.db_engine)
+        column_should_be_intlike(events_table, 'entity_id', self.db_engine)
+        column_should_be_timelike(events_table, 'outcome_date', self.db_engine)
+        column_should_be_booleanlike(events_table, 'outcome', self.db_engine)
+
+
+class StateConfigValidator(Validator):
+    def run(self, state_config):
+        if 'table_name' in state_config:
```

Comment:
maybe warn that will be falling back to default behavior if `table_name` isn't specified?